### PR TITLE
Feature/web api/connect to web api

### PR DESCRIPTION
**Pull request by IB-12**

### CODE REVIEW

**Comment by IB-12** [comment]
This allows us to make requests to the Web API using the AppRemote API's access token

File: app/src/main/java/ch/epfl/cs311/wanderwave/model/auth/AuthenticationController.kt [hunk 1]
```diff
@@ -74,28 +80,40 @@ constructor(
...
+      return false
+    }
+
+    return try {
+      val response = JSONObject(responseJson)
+      val firebaseToken = response.getString("firebase_token")
+      val spotifyAccessToken = response.getString("access_token")
+      val spotifyRefreshToken = response.getString("refresh_token")
+
+      Log.d("AuthenticationController", "SPOTIFY Access token: $spotifyAccessToken")
```
Comment (KlaKalma) [inline app/src/main/java/ch/epfl/cs311/wanderwave/model/auth/AuthenticationController.kt:94]: Is it a good thing to print this ? Seems like a security issue. I know it's just debug, but we should probably get rid of it now before we forget.

File: app/src/main/java/ch/epfl/cs311/wanderwave/model/auth/AuthenticationController.kt [hunk 2]
```diff
@@ -124,4 +142,31 @@ constructor(
...
+    // refresh the access token if necessary
+    if (!refreshSpotifyToken()) {
+      Log.d("AuthenticationController", "Failed to refresh Spotify token")
+      return "FAILURE"
+    }
+    // Get the access token from the AuthTokenRepository
+    val accessToken =
+        tokenRepository.getAuthToken(AuthTokenRepository.AuthTokenType.SPOTIFY_ACCESS_TOKEN)
+
+    Log.d("AuthenticationController", "Access token: $accessToken")
```
Comment (KlaKalma) [inline app/src/main/java/ch/epfl/cs311/wanderwave/model/auth/AuthenticationController.kt:158]: Same here

File: app/src/main/java/ch/epfl/cs311/wanderwave/model/auth/AuthenticationController.kt [hunk 3]
```diff
@@ -124,4 +142,31 @@ constructor(
   fun deauthenticate() {
     auth.signOut()
   }
+
+  suspend fun makeApiRequest(url: URL): String {
+    Log.d("AuthenticationController", "Making request to $url")
```
Comment (KlaKalma) [inline app/src/main/java/ch/epfl/cs311/wanderwave/model/auth/AuthenticationController.kt:147]: Same here

File: app/src/main/java/ch/epfl/cs311/wanderwave/model/auth/AuthenticationController.kt [hunk 4]
```diff
@@ -124,4 +142,31 @@ constructor(
...
+
+    // refresh the access token if necessary
+    if (!refreshSpotifyToken()) {
+      Log.d("AuthenticationController", "Failed to refresh Spotify token")
```
Comment (KlaKalma) [inline app/src/main/java/ch/epfl/cs311/wanderwave/model/auth/AuthenticationController.kt:151]: Log.e maybe

File: app/src/main/java/ch/epfl/cs311/wanderwave/model/auth/AuthenticationController.kt [hunk 5]
```diff
@@ -124,4 +142,31 @@ constructor(
...
+      return "FAILURE"
+    }
+    // Get the access token from the AuthTokenRepository
+    val accessToken =
+        tokenRepository.getAuthToken(AuthTokenRepository.AuthTokenType.SPOTIFY_ACCESS_TOKEN)
+
+    Log.d("AuthenticationController", "Access token: $accessToken")
+    return try {
+      (withContext(Dispatchers.IO) { url.openConnection() } as HttpURLConnection).run {
```
Comment (KlaKalma) [inline app/src/main/java/ch/epfl/cs311/wanderwave/model/auth/AuthenticationController.kt:160]: Use the global scope like yoric did before no ?

**Review by KlaKalma — CHANGES_REQUESTED** [review]
Quite good sir ! Soon the app will be perfect. Just a few changes

File: app/src/main/java/ch/epfl/cs311/wanderwave/model/auth/AuthenticationController.kt (hunk 1 above)
Comment (IB-12) [reply app/src/main/java/ch/epfl/cs311/wanderwave/model/auth/AuthenticationController.kt]: You're right, I forgot they were there

File: app/src/main/java/ch/epfl/cs311/wanderwave/model/auth/AuthenticationController.kt (hunk 5 above)
Comment (IB-12) [reply app/src/main/java/ch/epfl/cs311/wanderwave/model/auth/AuthenticationController.kt]: Yup, corrected it, thanks

**Review by KlaKalma — APPROVED** [review]
Perfect thanks

**Pull request merged**